srcCs/RTIDDSImpl.cs

```diff
@@ -1861,11 +1850,9 @@ public IMessagingWriter CreateWriter(string topic_name)
             }
 
 
-            if (("LatencyQos" == qos_profile ||
-                 "NoAckLatencyQos"  == qos_profile) &&
-                 !_DirectCommunication &&
-                (_Durability == 2 ||
-                 _Durability == 3)){
+            if ("LatencyQos" == qos_profile
+                    && !_DirectCommunication
+                    && (_Durability == 2 || _Durability == 3)){
```

Comment:
why dont we use:
- DDS::DurabilityQosPolicyKind::TRANSIENT_DURABILITY_QOS instead of 2
- DDS::DurabilityQosPolicyKind::PERSISTENT_DURABILITY_QOS  instead of 3

REF: https://community.rti.com/static/documentation/connext-dds/5.3.0/doc/api/connext_dds/api_dotnet/group__DDSDurabilityQosModule.html

Reply:
Done:

```
            if ("LatencyQos" == qos_profile
                    && !_DirectCommunication
                    && ((DDS.DurabilityQosPolicyKind)_Durability 
                            == DDS.DurabilityQosPolicyKind.TRANSIENT_DURABILITY_QOS
                        || (DDS.DurabilityQosPolicyKind)_Durability 
                            == DDS.DurabilityQosPolicyKind.PERSISTENT_DURABILITY_QOS))
            {
```